Library/Homebrew/test/dev-cmd/audit_spec.rb

```diff
@@ -217,6 +217,86 @@ class Foo < Formula
     end
   end
 
+  describe "#audit_deps" do
+    describe "a dependency on a macOS-provided keg-only formula" do
+      describe "which is whitelisted" do
+        let(:fa) do
+          formula_auditor "foo", <<-EOS.undent, new_formula: true
+            class Foo < Formula
+              url "http://example.com/foo-1.0.tgz"
+              homepage "http://example.com"
+
+              depends_on "openssl"
+            end
+          EOS
+        end
+
+        let(:fa_openssl) do
+          formula_auditor "openssl", <<-EOS.undent, new_formula: true
+            class Openssl < Formula
+              url "http://example.com/openssl-1.0.tgz"
+              homepage "http://example.com"
+
+              keg_only :provided_by_macos
+            end
+          EOS
+        end
+
+        before do
+          allow(fa.formula.deps[0])
+            .to receive(:to_formula).and_return(fa_openssl.formula)
```

Comment:
Not sure I understand why you have the `openssl` in `fa` rather than just with e.g. `formula do`?